### ISSUE TITLE
Support HTTP 1.0

### ISSUE THREAD

**Issue by kellabyte** [wg/wrk#339]
I was wondering if you could support HTTP 1.0 so that we can send out requests with the following change.

HTTP 1.1 requires the following.
```
GET / HTTP/1.1\r\nHost: 127.0.0.1:8000\r\n
```

HTTP 1.0 allows the following.
```
GET / HTTP/1.0\r\n
```

This allows us to double as many requests in the same amount of bytes. Maybe a `--http-version` argument?

**Comment by wg** [comment]
HTTP 1.0 doesn't use keep-alive by default so you'd need to add `Connection: keep-alive` which wouldn't help reduce the request size. However if your HTTP server isn't 100% HTTP 1.1 compliant and doesn't require Host you could simply create a Lua script that returns the appropriate HTTP request string without using `wrk.format`.

**Comment by kellabyte** [comment]
Oh I didn't know you could create totally custom strings with the lua scripting. Could you show me an example how? Avoiding having to modify wrk and doing it in a script would be ideal :)

**Comment by wg** [comment]
Yup the `request()` scripting method must return a string containing the HTTP request. Anything that can be parsed by http_parser is accepted and I don't think it cares about Host header correctness.

However overriding `request()` is most likely going to bottleneck on Lua function execution at the extremely high load levels you're targeting. Internally wrk performs some tricks to optimize the case of a static request, so this is undocumented behavior, but safe enough for now:

```
wrk.format = function()
   return "GET / HTTP/1.1\r\n\r\n"
end                  
```

**Comment by malanmurphy** [comment]
Thanks for the suggestion for creating an HTTP/1.0 request; it was a quick lifesaver in loadgen test I'm building to force 1.0 requests. I'm having an issue combining the above function with manual headers, though. This lua script works to force 1.0 but it's ignoring wrk.headers: 

`wrk.headers["user-agent"] = "My-UserAgent"
wrk.format = function()
   return "GET / HTTP/1.0\r\n\r\n"
end`

I've tried various combos of passing 'headers' into the function, reformatting the wrk.format function based on other example, etc, but can't figure out how to pass additional data (regardless of which wrk.* value I test) into a HTTP/1.0 request. Any help would be much appreciated. :)

Thx!

**Comment by wg** [comment]
@malanmurphy if you override `wrk.format` then you're responsible for building the entire HTTP request, including any headers =) You can use https://github.com/wg/wrk/blob/master/src/wrk.lua#L50 as a guide.
